tests/test_study_definition.py

```diff
@@ -56,11 +60,40 @@ def test_unrecognised_database_url_raises_error(tmp_path, monkeypatch):
         )
 
 
-def test_errors_are_triggered_without_database_url(tmp_path, monkeypatch):
+def test_errors_are_triggered_without_database_url(monkeypatch):
     monkeypatch.delenv("DATABASE_URL", raising=False)
     with pytest.raises(KeyError):
         StudyDefinition(
             population=patients.satisfying("no_such_column AND missing_column"),
             sex=patients.sex(),
             age=patients.age_as_of("2020-01-01",),
         )
+
+
+def test_pyodbc_not_accidentally_imported(tmp_path):
+    # Make a test script to be executed in a separate process so we can see
+    # what gets imported
+    def test_script():
+        import sys
+        from cohortextractor import StudyDefinition, patients
+
+        study = StudyDefinition(
+            population=patients.all(),
+            sex=patients.sex(
+                return_expectations={
+                    "rate": "universal",
+                    "date": {"earliest": "1900-01-01", "latest": "today"},
+                    "category": {"ratios": {"M": 0.49, "F": 0.51}},
+                }
+            ),
+        )
+        study.to_csv("/dev/null", expectations_population=10)
+        status = "is" if "pyodbc" in sys.modules else "is not"
+        print(f"pyodbc {status} imported")
+
+    # Convert the code in `test_script` to a standalone script by removing the
+    # function definition line and stripping the indentation
```

Comment:
Why not put the code in a string?

Reply:
Just because I like having it Blackened, linted etc and it was fairly easy to do